### ISSUE TITLE
Become very unstable in BERT+MultiTask mode

### ISSUE THREAD

**Issue by tanaka-jp** [LiyuanLucasLiu/RAdam#20]
我使用BERT+MultiTask来训练模型。
在每个train step的时候，task是随机选择的。

切换到RAdam后，train变得非常不稳定，有些task的loss变小了，但是有些task的loss变得非常大。
我觉得有可能是因为不同的task所产生的loss的大小不同，使得variance一直变得很大。
不知道我猜得对不对。

我想，像我这种情况，可能只用能Adam+WarmUp了。

**Comment by LiyuanLucasLiu** [comment]
emmm, bert 的训练比较tricky, 那个自带的adam 有一些trick (比如eps 是1e-6 而不是1e-8), 你可以试试RAdam + warmup，然后把这些trick 全加上。不过应该和好好tune 过的adam + warmup 差不多(如果差的多的话，warmup 也就不应该被解释成variance reduction 了). 

It's a little tricky to fine-tune/pre-train Bert. The original optimizer (Adam) is different from the conventional Adam (e.g., different eps, and in some implementations, the bias corrections are disabled). If you want to use RAdam, please make sure all these tricks are applied. At the same time, for a well-tuned warmup + learning rate scheduler, RAdam may not have much benefit (otherwise, warmup should not be interpreted as a variance reduction tech). As for the multi-task learning, i guess it needs a warmup stage due to the variance of grad.
